### PR TITLE
PDM updates

### PR DESCRIPTION
**Pull request by fatwir**
Modified the pyproject.toml file, created a pdm lock file and also included a requirements.txt file!

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+scipy==1.8.1
+scooby==0.10.0
+seaborn==0.13.2
+shellingham==1.5.4
+six==1.16.0
+sqlalchemy==2.0.35
+statsmodels==0.14.4
+tenacity==9.0.0
+threadpoolctl==3.5.0
+torch-scatter==2.1.2
```
Comment (ritvikvasan) [inline requirements.txt:62]: seems a bit off that torch is not a dependency? is this correct?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+drjit==0.4.6
+fonttools==4.54.1
+greenlet==3.1.1; (platform_machine == "win32" or platform_machine == "WIN32" or platform_machine == "AMD64" or platform_machine == "amd64" or platform_machine == "x86_64" or platform_machine == "ppc64le" or platform_machine == "aarch64") and python_version < "3.13"
+idna==3.10
+joblib==1.4.2
+kaleido==0.2.1.post1
+kiwisolver==1.4.7
+llvmlite==0.43.0
+markdown-it-py==3.0.0
+matplotlib==3.9.2
```
Comment (ritvikvasan) [inline requirements.txt:23]: I don't see monai?

**Review by ritvikvasan — COMMENTED** [review]
Looks good, but Im seeing some missing dependencies in the requirements file. Can someone help me understand how that works?

File: pyproject.toml [hunk 3]
```diff
@@ -23,23 +18,18 @@ dependencies = [
...
-    "seaborn>=0.13.2"
-]
-
-all = [
-    "br[viz]",
+    "seaborn>=0.13.2",
+    "plotly>=5.22.0"
 ]
+requires-python = "==3.10.*"
+license = {text = "MIT"}
```
Comment (pgarrison) [inline pyproject.toml:26]: We should leave out the non-dependency changes to `pyproject.toml`. E.g., the description, authors, and license

**Review by pgarrison — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (fatwir) [reply requirements.txt:62]: Yes, this is something I need to fix. In the updated .toml file, the build system flag has disappeared (torch needs to be included there to be included in the requirements.txt file ig - maybe @pgarrison can clarify this)

File: pyproject.toml (hunk 3 above)
Comment (fatwir) [reply pyproject.toml:26]: I do not fully understand this. Aren't these needed in the .toml file?

File: requirements.txt (hunk 2 above)
Comment (fatwir) [reply requirements.txt:23]: Yeah, I am not sure why this did not automatically come up in the requirements.txt file

File: pyproject.toml (hunk 3 above)
Comment (pgarrison) [reply pyproject.toml:26]: * The description text "Collection of methods for performing morphology appropriate representation learning" is better than the proposed "Default template for PDM package".
* The empty "{name = "", email = ""}," authorship line is unnecessary
* The license for this project is specified in the LICENSE file and is not MIT. (Beyond the scope of this PR, but the proper way to specify the license for this project is with `license = {file = "LICENSE"}`.)

File: requirements.txt (hunk 2 above)
Comment (pgarrison) [reply requirements.txt:23]: It appears this PR is not installing cyto-dl or its dependencies.

File: requirements.txt (hunk 1 above)
Comment (pgarrison) [reply requirements.txt:62]: I don't think it is related to the build system section of pyproject.toml: that section only specified numpy and pdm, nothing about torch. I think it's more likely related to leaving out cyto-dl

File: requirements.txt (hunk 2 above)
Comment (pgarrison) [reply requirements.txt:23]: @fatwir I think you had cyto-dl and its dependencies included in a commit yesterday? Let's talk more in person

File: requirements.txt (hunk 1 above)
Comment (fatwir) [reply requirements.txt:62]: I tried to install the env locally to see if that worked and pdm gave me an error citing that torch had to be included in the build section!

Comment (fatwir) [reply requirements.txt:62]: also, cyto-dl is included in the requirements.txt

**Pull request closed without merging**